src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSLlmModelProcessor.cs

```diff
@@ -0,0 +1,440 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
+using Amazon.Runtime;
+
+namespace OpenTelemetry.Instrumentation.AWS.Implementation;
+
+internal class AWSLlmModelProcessor
+{
+    internal static void ProcessRequestModelAttributes(Activity activity, AmazonWebServiceRequest request, string model)
```

Comment:
So there is a lot of duplicate code here that I think should be refactored. For example, `ProcessRequestModelAttributes` and `ProcessResponseModelAttributes` are the almost the same except the parameter list. Even the body extraction is the same: GetType -> GetProperty -> GetValue -> Process. Could you look on refactoring this code a bit more? Let me know what you think or if you want to discuss.

Reply:
I consolidated those two functions into one, as well as the model-specific functions. There wasn't a good base type shared by `AmazonWebServiceRequest` and `AmazonWebServiceResponse`, so I used a generic type. Let me know if approach is more preferable.